### ISSUE TITLE
Baudrate 250000 doesn't work under linux

### ISSUE THREAD

**Issue by SparkyCola** [pyserial/pyserial#129]
Hi, I've tried to use Baudrate 250000 under Linux (Ubuntu 16.10) with an Arduino-Mega-like Microcontroller (RamBo) and it doesn't connect.

**Comment by helarsen** [comment]
the error comes from the serialposix.py, about line 50:
'#set custom divisor'

 _buf[6] = buf[7] / baudrate_

_type(buf[6])_ is int, whereas _type(buf[7] / baudrate)_ is type float and this gives an exception in python3.4

To cure I have replaced it by 
_buf[6] = buf[7] // baudrate_

Or even better precision can be obtained by rounding instead. Like this
_buf[6] = (2*buf[7]+baudrate) // baudrate_

Hope this is the place to post this - if not let me know

**Comment by zsquareplusc** [comment]
This code was present in older versions (2.x). There was a patch that provided a different method to set the custom baudrate that is used now. If you really have this error and have installed 3.1.1, then I suspect you have an older version installed in parallel. I recommend to remove and other versions.

This is probably not the problem that the OP had, Ubuntu packages more recent versions.

**Comment by helarsen** [comment]
Thanks zsquareplusc, Yes you are completely right. I have no idea why python referenced this older version. I will try to get rid of it. How do I uninstall a specific python version.....?
apt-get remove python????

**Comment by zsquareplusc** [comment]
you do not need to uninstall a python installation, just duplicates of libraries. How to uninstall depends on how it was uninstalled. Sometimes .pyc files are left over.
if you want to use "pip install pyserial" i would suggest to "apt-get remove python-serial"

**Comment by diamondman** [comment]
@SparkyCola Any change this is all working for you now and we can get this issue closed out?
